xwiki-platform-core/xwiki-platform-display/xwiki-platform-display-html/src/main/java/org/xwiki/rendering/display/html/internal/DefaultTemplateHTMLDisplayer.java

```diff
@@ -0,0 +1,124 @@
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.display.html.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.script.ScriptContext;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.displayer.HTMLDisplayer;
+import org.xwiki.displayer.HTMLDisplayerException;
+import org.xwiki.script.ScriptContextManager;
+import org.xwiki.template.TemplateManager;
+
+/**
+ * Default implementation of {@code HTMLDisplayer} using templates.
+ *
+ * @version $Id$
+ * @since 10.11RC1
+ */
+@Component
+@Singleton
+public class DefaultTemplateHTMLDisplayer implements HTMLDisplayer
+{
+    /**
+     * Folder containing the HTML Displayers velocity templates.
+     */
+    public static final String TEMPLATE_FOLDER = "html_displayer";
+
+    /**
+     * Template extension (velocity).
+     */
+    public static final String TEMPLATE_EXTENSION = ".vm";
+
+    @Inject
+    protected TemplateManager templateManager;
+
+    @Inject
+    protected ScriptContextManager scriptContextManager;
+
+    /**
+     * {@inheritDoc}
+     *
+     * Displays the value with the 'view' mode.
+     */
+    @Override
+    public String display(Object value) throws HTMLDisplayerException
+    {
+        return display(value, new HashMap<>());
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Displays the value with the 'view' mode.
+     */
+    @Override
+    public String display(Object value, Map parameters) throws HTMLDisplayerException
+    {
+        return display(value, parameters, "view");
+    }
+
+    @Override
+    public String display(Object value, Map parameters, String mode) throws HTMLDisplayerException
+    {
+        try {
+            ScriptContext scriptContext = scriptContextManager.getCurrentScriptContext();
+            scriptContext.setAttribute("value", value, ScriptContext.ENGINE_SCOPE);
+            scriptContext.setAttribute("parameters", parameters, ScriptContext.ENGINE_SCOPE);
+            scriptContext.setAttribute("mode", mode, ScriptContext.ENGINE_SCOPE);
+
+            return templateManager.render(getTemplateName(value, mode));
+        } catch (Exception e) {
+            throw new HTMLDisplayerException("Couldn't render the template", e);
+        }
+    }
+
+    /**
+     * Computes the template name.
+     * The following names will be use in this priority order to find an existing template:
+     * <ul>
+     *   <li>html_displayer/[type]/[mode].vm
+     *   <li>html_displayer/[type].vm
+     *   <li>html_displayer/[mode].vm
+     *   <li>html_displayer/default.vm
+     * </ul>
+     * @return the template name used to make the rendering
+     */
+    protected String getTemplateName(Object value, String mode) {
```

Comment:
I think it would make more sense in term of performance to have a `Template getTemplate(Object value, String mode)` instead of redoing the template search from the name twice.

Reply:
The idea was to override the method to change the logic on how the name is computed. I guess returning the `Template` instead of its name won't hurt.

Reply:
Also does it really need to be protected ? DefaultTemplateHTMLDisplayer seems to be used more as a component than a super class.

Reply:
Actually it's not used yet, I trough I had seen it used in another class.

Reply:
If you want to override it then yes. It's true that DefaultTemplateHTMLDisplayer is not meant to be overriden *anymore* (it was an abstract class at first). I think you are right, I'll make the method `private` and make it return a `Template`.